client/client.go

```diff
@@ -41,8 +43,8 @@ func init() {
 
 // Configuration is the client configuration
 type Configuration struct {
-	// AuthHeader is the header we should use to make API calls.
```

Comment:
This is the main story of this PR.

We replace `AuthHeader string` with a `AuthHeaderGetter` that returns a string.

That gives us an opportunity to run checks against the AuthHeader before it gets used.